scripts/process.py

```diff
@@ -3,56 +3,33 @@
 to load an xml file into a sqla object with all the correct keys.     
 """
 
-from ukrdc_cupid.core.parse.utils import load_xml_from_path
-from ukrdc_cupid.core.match.identify import link_patient_number 
-from ukrdc_cupid.core.store.models.ukrdc import PatientRecord
+import glob
+import os
 
-from sqlalchemy.orm import sessionmaker
 from xsdata.exceptions import ParserError
-from sqlalchemy import create_engine
-
-from ukrdc_cupid.core.store.keygen import mint_new_pid
-    
-#engine = Connection.get_engine_from_file(key="ukrdc_staging")
-#session = scoped_session(sessionmaker(engine))
-url = "postgresql://postgres:postgres@localhost:5432/dummy_ukrdc"
-#engine = create_engine(url, echo = True)
-engine = create_engine(url)
-
-ukrdc3_sessionmaker = sessionmaker(bind=engine)
-session = ukrdc3_sessionmaker()
-
-# load xml file as python object 
-#xml_file = r"Q:\UKRDC\UKRDC Feed Development\RFBAK Leicester\RFBAK_00082_4165311820.xml"
-xml_file = r"scripts/xml_examples/UKRDC.xml"
-xml_object = load_xml_from_path(xml_file)
-
-if isinstance(xml_object, ParserError):
-    print(f"File load failed for the following reason: {xml_object}")
-else: 
-    # link patient to existing record 
-    linked_patients = link_patient_number(session=session, xml=xml_object)
-
-    if len(linked_patients) == 0: 
-        # create new pid and load into sqla
-        pid = mint_new_pid(session=session)
-        patient_record = PatientRecord(xml_object)
-        patient_record.map_xml_to_tree()
-        patient_record.transform(pid)
-        pr_orm = patient_record.get_orm_list()
-        print(pr_orm)
-
-    elif len(linked_patients) == 1:
-        # assign existing pid and ukrdcid 
-        patient_record = PatientRecord(xml_object)
-
-        
-    else:
-        # if multiple matches are returned don't attempt to choose one
-        print("Patient cannot be unambigously matched to multiple records")
-
-
+from ukrdc_cupid.core.utils import DatabaseConnection
+from ukrdc_cupid.core.parse.utils import load_xml_from_path
+from ukrdc_cupid.core.general import process_file
 
+session = DatabaseConnection(prefix="UKRDC").create_session()
+ 
+# Specify the directory where your XML files are located
+xml_directory = ".xml_to_load"
 
+# grab files to load from directory
+xml_files = glob.glob(os.path.join(xml_directory, "*.xml"))
 
 
+for xml_file in xml_files:
+    xml_object = load_xml_from_path(xml_file)
+    if isinstance(xml_object, ParserError):
+        raise f"File load failed for the following reason: {xml_object}"
+    
+    investigation = process_file(xml_object, session)
+
+    if investigation:
```

Comment:
Not quite sure what this is doing - but inanycase what happens if more than one files in the loop triggers an investigation?